etomica-apps/src/main/java/etomica/freeenergy/npath/ConfigFromFileLAMMPS.java

```diff
@@ -191,24 +191,20 @@ public String getLabel() {
         filterSlider.setMaximum(2);
         filterSlider.setPrecision(2);
         filterSlider.setNMajor(4);
-        filterSlider.setPostAction(new IAction() {
-
-            @Override
-            public void actionPerformed() {
-                graphic.getDisplayBox(box).repaint();
-            }
-        });
+        Box finalBox = box;
+        filterSlider.setPostAction(() -> graphic.getDisplayBox(finalBox).repaint());
         graphic.add(filterSlider);
         DeviceSlider configSlider = new DeviceSlider(sim.getController(), modifierConfig);
         configSlider.setMaximum(config - 1);
         configSlider.setNMajor(config);
         configSlider.setMinimum(0);
         configSlider.setPrecision(0);
+        Box finalBox1 = box;
```

Comment:
Why do you need another (third!) Box variable?

Reply:
I don't I could just use the first one, I was just using the quickfix. There has to be one final reference though because you can only use local variables in anonymous classes if they're effectively final, so once I had it reassign box it became a compilation error to use it in the Action.